frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -142,8 +123,46 @@ const AuthWrapper: React.FC<AuthWrapperProps> = ({
       if (refreshTokenInterval.current)
         clearInterval(refreshTokenInterval.current);
     };
-  }, [tokenResponse?.refreshToken, tokenResponse?.expiresIn]);
+  }, [
+    tokenResponse?.refreshToken,
+    tokenResponse?.expiresIn,
+    JSON.stringify(discovery),
+  ]);
+
+  // Run onTokenChange callback
+  // store token in session storage or remove it when undefined
+  useEffect(() => {
+    if (!tokenResponse) {
+      onTokenChange('');
+      sessionStorage.removeItem(injectToken);
+    } else {
+      const token = (() => {
+        switch (injectToken) {
+          case 'access_token':
+            return tokenResponse?.accessToken ?? '';
+          case 'id_token':
+            return tokenResponse?.idToken ?? '';
+          default:
+            return '';
+        }
+      })();
+      onTokenChange(token);
+      sessionStorage.setItem(injectToken, JSON.stringify(tokenResponse));
+    }
+  }, [tokenResponse?.accessToken]);
```

Comment:
I think the deps here should be `JSON.stringify(tokenResponse)` because of the line `sessionStorage.setItem(injectToken, JSON.stringify(tokenResponse))` and if `injectToken` was `id_token`